### PR TITLE
[Framework] allow to create kernel-aware console applications without default commands

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| License       | MIT
| Doc PR        |  https://github.com/symfony/symfony-docs/pull/7296

Suppose that you need to create a kernel-aware application:

```
// ...
$kernel = new AppKernel($env, $debug);
$application = new MyApplication($kernel);

$application->addCommands([
    // CUstom commands
    new MyCustomCommand(),
]);

$application->run($input);
```
The code above will create a console application with  ``MyCustomCommand`` plus all the other default symfony commands.

With this PR you are able to specify whether or not you want to add the default commands:

``$application = new MyApplication($kernel, false);``
or
``$application = new MyApplication($kernel, true);`` (default)

### CODE REVIEW

**Comment by ShinDarth** [comment]
@ro0NL  here

**Comment by ro0NL** [comment]
You could have changed the base branch on your previous PR as well ;-)

File: src/Symfony/Bundle/FrameworkBundle/Console/Application.php [hunk 1]
```diff
@@ -33,11 +33,13 @@ class Application extends BaseApplication
...
      *
-     * @param KernelInterface $kernel A KernelInterface instance
+     * @param KernelInterface $kernel              A KernelInterface instance
+     * @param bool            $loadDefaultCommands Load all the default commands
      */
-    public function __construct(KernelInterface $kernel)
+    public function __construct(KernelInterface $kernel, $loadDefaultCommands = true)
     {
         $this->kernel = $kernel;
+        $this->commandsRegistered = !$loadDefaultCommands;
```
Comment (HeahDude) [inline src/Symfony/Bundle/FrameworkBundle/Console/Application.php]: You can remove the initialization of `$this->commandsRegistered` above though.

Comment (ShinDarth) [inline src/Symfony/Bundle/FrameworkBundle/Console/Application.php]: done

**Comment by iltar** [comment]
This should make it easier to create single command applications I believe?

**Comment by ShinDarth** [comment]
@iltar yes, in particular kernel-aware ones

**Review by mickaelandrieu — APPROVED** [review]

**Comment by ShinDarth** [comment]
Thanks all for reviewing this.

Now about the doc part, is there already a page which explains how to create kernel-aware console applications? 

Afaik the only place where ``Symfony\Bundle\FrameworkBundle\Console\Application`` is mentioned is for [testing commands](http://symfony.com/doc/current/console.html#testing-commands).

**Comment by ro0NL** [comment]
Maybe a special note at https://symfony.com/doc/current/components/console/single_command_tool.html? Howto create a single command app with  the framework.. something like that?

**Comment by ShinDarth** [comment]
@ro0NL I don't think so, because it hasn't to be one single command. It can be even with multiple commands.

Maybe it's better to create a new dedicated page about "How to create a kernel-aware console application" ?

**Comment by ro0NL** [comment]
It's about creating awareness of this small feature. I think the single command tool page has the most visits by now, and is best indexed.

Besides, this definitely relates to creating a _truly_ single command tool with the framework ;-) If we explain it right there, people will probably understand this is applies to multiple commands as well.

**Comment by ShinDarth** [comment]
PR for the doc done: 

https://github.com/symfony/symfony-docs/pull/7296

**Comment by ShinDarth** [comment]
any news?

File: src/Symfony/Bundle/FrameworkBundle/Console/Application.php [hunk 2]
```diff
@@ -28,16 +28,18 @@
...
 
     /**
      * Constructor.
      *
-     * @param KernelInterface $kernel A KernelInterface instance
+     * @param KernelInterface $kernel              A KernelInterface instance
+     * @param bool            $loadDefaultCommands Load all the default commands
      */
-    public function __construct(KernelInterface $kernel)
+    public function __construct(KernelInterface $kernel, $loadDefaultCommands = true)
```
Comment (lyrixx) [inline src/Symfony/Bundle/FrameworkBundle/Console/Application.php]: I would name it `registerDefaultCommands` to be consistent.

**Comment by nicolas-grekas** [comment]
Moving to 4.1. Rebase on master needed, where PHP 7.1 features can be used btw.

**Comment by OskarStark** [comment]
@FrancescoBorzi can you please rebase this PR?

**Comment by FrancescoBorzi** [comment]
I opened this PR on `22 Dec 2016` so almost 3 years ago.

I no longer use Symfony for a while and to be honest I don't even remember what I did here.

Sorry.

**Pull request closed without merging**